endpoints/books/controller.py

```diff
@@ -33,38 +36,49 @@ def post(self):
         try:
             current_user = get_jwt_identity()
             book = request.get_json()
-            book['user_id'] = current_user
+            existing_book = userBookRepository.findByName(book['name'])
+            if existing_book['name'] == book['name'] and existing_book['user_id'] == current_user['id']:
+                return {'message': 'Resource already exists', 'time': datetime.datetime.now().isoformat()}, 422
+            book['user_id'] = current_user['id']
             userBookRepository.add(book)
             return marshal(book, book_list_fields)
 
         except Exception as error:
-            return {'message': 'Something went wrong', 'timestamp': round(time.time())}, 500
-            
+            logging.error(f'{error}')
+            return {'message': 'Something went wrong', 'time': datetime.datetime.now().isoformat()}, 500
+
     @jwt_required
     def get(self):
         try:
-            book = userBookRepository.find()
+            current_user = get_jwt_identity()
+            book = userBookRepository.findByUser(current_user['id'])
             return marshal(book, book_list_fields)
-
         except Exception as error:
-            return {'message': 'Something went wrong', 'timestamp': round(time.time())}, 500
+            logging.error(f'{error}')
+            return {'message': 'Something went wrong', 'time': datetime.datetime.now().isoformat()}, 500
 
 
 class UserBookByIdResource(Resource):
     @jwt_required
     def get(self, id):
         try:
+            current_user = get_jwt_identity()
+            if current_user['id'] != id and not current_user['admin']:
```

Comment:
You can probably move this to a decorator. Also, does it matter that is an admin for the sake of getting the book information?

Reply:
I'm just making that an admin account can access to the book information.